src/NServiceBus.Core/obsoletes.cs

```diff
@@ -693,19 +694,19 @@ public ICallback Send<T>(Address address, string correlationId, Action<T> messag
         }
 
         [ObsoleteEx(
-            ReplacementTypeOrMember = "ISendOnlyBus.Publish(object message, PublishOptions options)",
+            ReplacementTypeOrMember = "ISendOnlyBus.PublishAsync(object message, PublishOptions options)",
             RemoveInVersion = "7.0",
             TreatAsErrorFromVersion = "6.0")]
-        public void Publish(object message, NServiceBus.PublishOptions options)
+        public Task PublishAsync(object message, NServiceBus.PublishOptions options)
```

Comment:
Shouldn't we keep the old signatures and have them point to the new async versions? (cc @SimonCropp )

That way the users would get a good compiler error guiding them?